tests/test_qaoa_objective_maxcut.py

```diff
@@ -0,0 +1,35 @@
+###############################################################################
+# // SPDX-License-Identifier: Apache-2.0
+# // Copyright : JP Morgan Chase & Co
+###############################################################################
+import networkx as nx
+import numpy as np
+from qokit.qaoa_objective_maxcut import get_qaoa_maxcut_objective
+from qokit.utils import precompute_energies
+from qokit.parameter_utils import get_fixed_gamma_beta
+from qokit.maxcut import maxcut_obj, get_adjacency_matrix
+
+
+def test_validate_energy_for_terms_and_precomputedcuts_are_same():
+    G = nx.random_regular_graph(3, 16)
+    # without precomputed cuts
+    f_terms = get_qaoa_maxcut_objective(G.number_of_nodes(), 1, G=G, parameterization="gamma beta")
+    # with precomputed cuts
+    precomputed_cuts = precompute_energies(maxcut_obj, G.number_of_nodes(), w=get_adjacency_matrix(G))
+    f_precomputedcuts = get_qaoa_maxcut_objective(G.number_of_nodes(), 1, precomputed_cuts=precomputed_cuts, parameterization="gamma beta")
+    p = 1
+    gamma, beta = get_fixed_gamma_beta(3, p)
+    energy_terms = f_terms(-1 * np.asarray(gamma), beta)
```

Comment:
@rsln-s  I would like to confirm if flipping the sign will have any side-effects for other use cases? for e.g. for max cut we are reversing the sign for maximization problem , if any other use case is minimization problem. Would it not affect the results?

Reply:
`get_fixed_gamma_beta` is only used for MaxCut; I don't believe there would be any side-effects.

Reply:
@rsln-s Some of the test cases use sample_from_weighted_Shaydulin_Lotshaw_2022.json to get beta and gamma values. Shall I go ahead to change the sign in this file?
get_sk_gamma_beta is also used in Portfolio Optimization code. Changing the sign will force sign change in those code. Either we keep same and change sign in the test case or we can create 2 methods seperately to cater for minimization and maximization respectively

Reply:
@ZichangHe, do we assume minimization or maximization for QAOA objective for portfolio optimization? We should make sure it's minimization to ensure consistency

Reply:
We assumed minimization in portfolio optimization as well. BTW, I donot think flip the sign of gamma would translate a minimal to maximal? They are not guaranteed to be symmetric?

Reply:
Would flipping the sign break portfolio opt? Do we even test portfolio opt currently?..

Reply:
There is a test making use of `get_sk_ini ` https://github.com/jpmorganchase/QOKit/blob/4850f358bd594da8db4697725d9386f99b093dd4/tests/test_portfolio_optimization.py#L133C13-L133C13

Reply:
@rsln-s @ZichangHe  There are test cases for Portfolio Optimization -->tests/test_portfolio_optimization.py::test_portfolio_AR this test case fails because of get_sk_gamma_beta sign switch

Reply:
@rsln-s My suggestion for sample_from_weighted_Shaydulin_Lotshaw_2022.json this is read the file and then change the sign within test case before testing. This is will be easier.
For get_sk_gamma_beta we leave the sign as is and make the sign switch in max_cut.

Reply:
@rajgane07 I'm OK with the proposed solution.
@ZichangHe, would you mind checking what's up with the sign in portfolio opt? It's a bit surprising that even though we're minimizing, gamma is used with the wrong sign

Reply:
If the sign in `get_sk_gamma_beta` is flipped, we can make the change by flip the `gamma_scale` here: https://github.com/jpmorganchase/QOKit/blob/4850f358bd594da8db4697725d9386f99b093dd4/qokit/portfolio_optimization.py#L274

The portfolio setup should be fine.